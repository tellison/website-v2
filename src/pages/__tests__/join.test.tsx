import React from 'react';
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest'
import { axe } from 'vitest-axe';
import Join, { Head } from '../join';

describe('Join page', () => {
  it('renders correctly', () => {
    const { container } = render(<Join />);
    // eslint-disable-next-line
    const pageContent = container.querySelector('main');

    expect(pageContent).toMatchSnapshot();
  });

  it('head renders correctly', () => {
    const { container } = render(<Head />);
    // eslint-disable-next-line
    const title = container.querySelector('title');
    expect(title?.textContent).toEqual('Join | Adoptium');
  });

  it('has no accessibility violations', async () => {
    const { container } = render(<Join />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
