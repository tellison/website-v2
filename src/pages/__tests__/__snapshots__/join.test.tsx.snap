// Vitest Snapshot v1

exports[`Join page > renders correctly 1`] = `
<main>
  <section
    class="py-5 text-center container"
  >
    <div
      class="row py-lg-5"
    >
      <div
        class="col-lg-8 col-md-8 mx-auto"
      >
        <h1
          class="fw-light"
        >
          Join the Eclipse AdoptiumÂ® Working Group
        </h1>
      </div>
      <div>
        <div
          id="reactHubspotForm0"
          style="display: none;"
        />
      </div>
    </div>
  </section>
</main>
`;
